pkg/controller/backup-operator/abs_backup.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"crypto/tls"
+	"fmt"
+
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/backup"
+	"github.com/coreos/etcd-operator/pkg/backup/writer"
+	"github.com/coreos/etcd-operator/pkg/util/azureutil/absfactory"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+// TODO: replace this with generic backend interface for other options (PV, Azure)
+// handleABS saves etcd cluster's backup to specificed ABS path.
+func handleABS(kubecli kubernetes.Interface, s *api.ABSBackupSource, endpoints []string, clientTLSSecret, namespace string) (*api.BackupStatus, error) {
+	cli, err := absfactory.NewClientFromSecret(kubecli, namespace, s.ABSSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	var tlsConfig *tls.Config
+	if len(clientTLSSecret) != 0 {
```

Comment:
This code is duplicate to handleS3(). Can you refactor it?

Reply:
I think we can just extract line 39 - 49 to something like `generateTLSConfig` ?